pkg/workloads/console/controller.go

```diff
@@ -316,21 +319,29 @@ func (r *reconciler) setConsoleOwner(consoleTemplate *workloadsv1alpha1.ConsoleT
 }
 
 func (r *reconciler) setConsoleTTL(consoleTemplate *workloadsv1alpha1.ConsoleTemplate) error {
-	if r.console.Spec.TTLSecondsAfterFinished != nil {
-		return nil
-	}
+	defaultTTLSecondsBeforeRunning := int32(DefaultTTLBeforeRunning.Seconds())
+	defaultTTLSecondsAfterFinished := int32(DefaultTTLAfterFinished.Seconds())
 
 	updatedCsl := r.console.DeepCopy()
-	defaultTTL := int32(DefaultTTL.Seconds())
 
-	if consoleTemplate.Spec.DefaultTTLSecondsAfterFinished != nil {
+	if r.console.Spec.TTLSecondsBeforeRunning != nil {
+		updatedCsl.Spec.TTLSecondsBeforeRunning = r.console.Spec.TTLSecondsBeforeRunning
+	} else if consoleTemplate.Spec.DefaultTTLSecondsBeforeRunning != nil {
+		updatedCsl.Spec.TTLSecondsBeforeRunning = consoleTemplate.Spec.DefaultTTLSecondsBeforeRunning
+	} else {
+		updatedCsl.Spec.TTLSecondsBeforeRunning = &defaultTTLSecondsBeforeRunning
+	}
+
+	if r.console.Spec.TTLSecondsAfterFinished != nil {
+		updatedCsl.Spec.TTLSecondsAfterFinished = r.console.Spec.TTLSecondsAfterFinished
+	} else if consoleTemplate.Spec.DefaultTTLSecondsAfterFinished != nil {
 		updatedCsl.Spec.TTLSecondsAfterFinished = consoleTemplate.Spec.DefaultTTLSecondsAfterFinished
 	} else {
-		updatedCsl.Spec.TTLSecondsAfterFinished = &defaultTTL
+		updatedCsl.Spec.TTLSecondsAfterFinished = &defaultTTLSecondsAfterFinished
 	}
 
 	if err := r.client.Update(r.ctx, updatedCsl); err != nil {
```

Comment:
Because we've dropped the early return, are we now always doing updates to the console whenever we go through the reconcile loop?

Reply:
Ah yep this is totally true!

Reply:
Discussed: We'll try replacing this with a call to CreateOrUpdate; otherwise re-add the conditional at the top and check both fields